packages/polyfill/source/tests/selectors.test.ts

```diff
@@ -0,0 +1,280 @@
+import {Window} from '../index.ts';
+import {parseSelector} from '../selectors.ts';
+
+import {describe, it, expect, beforeEach} from 'vitest';
+
+describe('selector parsing and matching', () => {
+  beforeEach(() => {
+    const window = new Window();
+    Window.setGlobalThis(window);
+  });
+
+  describe('parseSelector', () => {
+    it('parses element selectors', () => {
+      const parts = parseSelector('div');
+      expect(parts).toHaveLength(1);
+      expect(parts[0]!.matchers).toHaveLength(1);
+      expect(parts[0]!!.matchers[0]!).toMatchObject({
```

Comment:
Are the double `!!.` intended here throughout this file? I was surprised to see this even compiled 😅